src/Raven.Server/Documents/Queries/QueryMetadata.cs

```diff
@@ -97,6 +92,14 @@ public QueryMetadata(Query query, BlittableJsonReaderObject parameters, ulong ca
             DeclaredFunctions = Query.DeclaredFunctions;
 
             Build(parameters);
+            
+            if (query.Filter != null)
+            {
+                if (HasVectorSearch)
+                    throw new InvalidQueryException("Cannot use 'filter' when 'vector.search' is used in where statement.");
+                
+                BuildFilterScript(query);
```

Comment:
Just wondering why it got moved here. Is it because `HasVectorSearch` is set during `Build()`?